plugins/cachix/auth_token.go

```diff
@@ -33,9 +36,41 @@ func AuthToken() schema.CredentialType {
 		DefaultProvisioner: provision.EnvVars(defaultEnvVarMapping),
 		Importer: importer.TryAll(
 			importer.TryEnvVarPair(defaultEnvVarMapping),
+			TryCachixConfigFile(),
 		)}
 }
 
 var defaultEnvVarMapping = map[string]sdk.FieldName{
 	"CACHIX_AUTH_TOKEN": fieldname.Token,
 }
+
+func TryCachixConfigFile() sdk.Importer {
+	return importer.TryFile("~/.config/cachix/cachix.dhall", func(ctx context.Context, contents importer.FileContents, in sdk.ImportInput, out *sdk.ImportAttempt) {
+		var token string
+
+		fileString := contents.ToString()
+		// The dhall config file contains a single block with variables inside
+		fileString = strings.Trim(fileString, "{}")
+		// Remove spaces to make parsing simpler
+		fileString = strings.Join(strings.Fields(fileString), "")
+		// There should be 3 fields: authToken, hostname, binaryCaches
+		keyVals := strings.SplitN(fileString, ",", 3)
```

Comment:
I propose we remove the assumption that the file will have 3 fields, so that if Cachix chooses to extend the config file in the future with an extra field, this doesn't break the importer.

```suggestion
		keyVals := strings.Split(fileString, ",")
```

Reply:
Thanks for the suggestion, I've made the change. 😊